.github/workflows/mypy-clinic.yml

```diff
@@ -0,0 +1,37 @@
+name: Run mypy on Tools/clinic
```

Comment:
Rather than having this as a "mypy on clinic" job, should it be a general mypy job?

Might we want to check other things with mypy?

Reply:
Maybe. But not right now :)

We can always change the name later if we add other things, right?

Reply:
pegen would be a good candidate, probably. But agree that's out of scope.

Reply:
Yeah, we can rename later, and add more later.

Although the old names still show in the list on the left at https://github.com/python/cpython/actions/workflows/ and it'd be nice to avoid a bit of clutter, if a more generic _name_ is fine now.

Reply:
Fair point, I'll go with a shorter name!